src/elements/List/ListList.js

```diff
@@ -0,0 +1,42 @@
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+
+import {
+  customPropTypes,
+  getElementType,
+  getUnhandledProps,
+  META,
+  useKeyOnly,
+} from '../../lib'
+
+function ListList(props) {
+  const { children, className } = props
+
+  const ElementType = getElementType(ListList, props)
+  const classes = cx(
+    className,
+    useKeyOnly(ElementType !== 'ul' && ElementType !== 'ol', 'list'),
```

Comment:
Oh boy, well then.  I think the logic you originally had there is just the way it has to be.